priv/templates/phx.gen.json/controller_test.exs

```diff
@@ -37,7 +37,7 @@ defmodule <%= inspect context.web_module %>.<%= inspect Module.concat(schema.web
 
       assert %{
                "id" => id<%= for {key, val} <- schema.params.create do %>,
-               "<%= key %>" => <%= Phoenix.json_library().encode!(val) %><% end %>
+               "<%= key %>" => <%= if val == %{}, do: "%{}", else: Phoenix.json_library().encode!(val) %><% end %>
```

Comment:
```suggestion
               "<%= key %>" => <%= if val == %{}, do: "{}", else: Phoenix.json_library().encode!(val) %><% end %>
```

Probably I'm missing something, but empty map in json is `{}`, not `%{}` (which is Elixir map).

Reply:
The patch is fine, it's about generating _Elixir_ code, for `key = "foo"` and `val = %{}` it would generate:

```elixir
assert
  # ...
  "foo" => %{}
```

which is correct.

Since it's about generating Elixir map, another way would be to do:

```elixir
val |> Phoenix.json_library().encode!() |> Phoenix.json_library().decode!() |> inspect()
```

which is probably a bit too much though :)

Reply:
@AndrewDryga as @wojtekmach mentioned it's about generating asserts with `json_response` which decodes json body to elixir structures.

@wojtekmach yeah your code is common. Keep PR as is or update?)

Reply:
So, guys, what do think about this PR?